### PR TITLE
Feat: publish package to PyPI

### LINKED ISSUE #22: Publish to PyPI

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #22

## Overview

This PR updates our package configuration files (`setup.cfg` and `pyproject.toml`) and introduces a workflow to publish to PyPI when we make a new release on GitHub.

## Other work done
API tokens (scoped to `eligibility-api`) have been created on PyPI/TestPyPI and added to this GitHub repository as secrets so that the workflow can handle our future releases.

For this initial release, I installed `build` and `twine` in the devcontainer and created short-lived API tokens on PyPI/TestPyPI to upload the distribution files.

See the published package at:
 - PyPI: https://pypi.org/project/eligibility-api/ 
 - TestPyPI: https://test.pypi.org/project/eligibility-api/
    - (Note: I was testing some changes to `setup.cfg` and had to upload with a locally-bumped version since you can't reupload with the same name)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          python-version-file: .github/workflows/.python-version
+
+      - name: Install build dependencies
+        run: pip install build
+
+      - name: Build package
+        run: python -m build
+
+      - name: Publish to Test PyPI
```
Comment (thekaveman) [inline .github/workflows/publish.yml:25]: We need some additional parameters to correctly publish to Test PyPI. 

See https://github.com/pypa/gh-action-pypi-publish#advanced-release-management

And for example: https://github.com/compilerla/conventional-pre-commit/blob/main/.github/workflows/publish.yml#L26

**Review by thekaveman — CHANGES_REQUESTED** [review]
One change needed to publish to Test PyPI.

As you saw, you can't overwrite existing versions/tags. It seems like the ones already up there are using the format `YYYY.M.D` whereas this PR uses `YYYY.MM.D`, so the publish should succeed when you make a release here in GitHub.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (angela-tran) [reply .github/workflows/publish.yml:25]: Ah, thanks for catching that. Added the missing parameters (and fixed the `user` parameter) in 3fd33a1c924d12cc936205b1b042cb6371492cf5

**Comment by angela-tran** [comment]
Regarding `YYYY.MM.R` vs. `YYYY.M.R`, I noticed that `setuptools` (the backend that we've configured `build` to use) is normalizing the version, i.e. stripping the leading zero.

There was a long discussion on https://github.com/pypa/setuptools/issues/308 about it as well as a [PR ](https://github.com/pypa/setuptools/pull/2026) that introduced a way to bypass the normalization. When I tried using the `sic` function and built our package locally, it runs using `egg_info`, which [doesn't work with `sic` currently](https://github.com/pypa/setuptools/issues/2181).

```
...
running sdist
running egg_info
writing eligibility_api.egg-info/PKG-INFO
writing dependency_links to eligibility_api.egg-info/dependency_links.txt
writing requirements to eligibility_api.egg-info/requires.txt
writing top-level names to eligibility_api.egg-info/top_level.txt
...
```

It was taking too long to try and fix this, so this is as far as I got with looking into it.

**Comment by thekaveman** [comment]
> Regarding YYYY.MM.R vs. YYYY.M.R, I noticed that setuptools (the backend that we've configured build to use) is normalizing the version, i.e. stripping the leading zero.

Interesting... so that means that we can't actually publish this version with the workflow, since it already exists on PyPI right?

**Comment by angela-tran** [comment]
> Interesting... so that means that we can't actually publish this version with the workflow, since it already exists on PyPI right? (That's fine, we'll just have to push another version/change here before making a release that publishes to PyPI)

Yep, that's my understanding of it all.

**Review by thekaveman — APPROVED** [review]

**Pull request merged**